oldabe/money_in.py

```diff
@@ -504,22 +519,26 @@ def get_amounts_owed(total_amount, attributions):
             for email, share in attributions.items()}
 
 
-def distribute_remaining_amount(remaining_amount, truncated_payable_attributions, payment):
+def redistribute_pot(redistribution_pot, attributions, unpayable_contributors, payment_file, amounts_owed):
     """
-    After paying off debts, distribute remaining amount to payable contributors
-    in the attributions file, by renormalizing after excluding unpayable
-    contributors.
+    Redistribute the pot of remaining money over all payable contributors, according to attributions
+    share (normalized to 100%). Create advances for those amounts (because they are in excess
+    of the amount owed to each contributor from the original payment) and add the amounts to the
+    amounts_owed dictionary to keep track of the full amount we are about to pay everyone.
     """
-    amounts_owed = get_amounts_owed(remaining_amount, truncated_payable_attributions)
-    remainder_attributions = normalize(truncated_payable_attributions)
-    advances
-    transactions = generate_transactions(
-        remaining_amount, remainder_attributions, payment.file, commit_hash
+    fresh_advances = []
+    normalized_payable_attributions = normalize(
+        {email: share for email, share in attributions.items() if email not in unpayable_contributors}
     )
+    for email, share in normalized_payable_attributions.items():
+        advance_amount = redistribution_pot * share
+        fresh_advances.append(Advance(email=email,
+                                      amount=advance_amount,
+                                      payment_file=payment_file,
+                                      commit_hash=get_git_revision_short_hash()))
+        amounts_owed[email] += advance_amount
 
-    commit_hash = get_git_revision_short_hash()
-
-    return transactions
+    return fresh_advances
```

Comment:
This function mutates the `amounts_owed` dictionary and also returns the freshly created advances. Would it be better to instead pass in the (precomputed) payable normalized attributions and the amounts_owed as inputs, and then use the returned advances to update the `amounts_owed` dictionary after this returns (maybe even in a separate dedicated function)? In that case, the only deliverable for this function would be the created advances. To reflect the narrow focus, we could call this function `issue_advances`.

Reply:
I kind of like that this function encapsulates the business logic of redistributing the pot in one place actually.